grype/pkg/package.go

```diff
@@ -44,7 +45,7 @@ type Package struct {
 	Metadata  interface{} // This is NOT 1-for-1 the syft metadata! Only the select data needed for vulnerability matching
 }
 
-func New(p syftPkg.Package) Package {
+func New(p syftPkg.Package, enhancers ...enhancer) Package {
```

Comment:
why make an unexposed type that's in a signature of a function in the public api? Though it's still usable it's a little awkward to discover. Any reason not to use the raw function definition here?